core/store/orm/orm.go

```diff
@@ -485,24 +485,58 @@ func (orm *ORM) AnyJobWithType(taskTypeName string) (bool, error) {
 	return found, ignoreRecordNotFound(rval)
 }
 
-// CreateTx saves the properties of an Ethereum transaction to the database.
+// CreateTx returns a transaction by its surrogate key, if it exists, or
+// creates it and its attempts
 func (orm *ORM) CreateTx(
-	from common.Address,
-	nonce uint64,
-	to common.Address,
-	data []byte,
-	value *big.Int,
-	gasLimit uint64,
+	surrogateID null.String,
+	ethTx *types.Transaction,
+	from *common.Address,
+	sentAt uint64,
 ) (*models.Tx, error) {
-	tx := models.Tx{
-		From:     from,
-		To:       to,
-		Nonce:    nonce,
-		Data:     data,
-		Value:    models.NewBig(value),
-		GasLimit: gasLimit,
-	}
-	return &tx, orm.DB.Save(&tx).Error
+	var tx models.Tx
+	err := orm.DB.
+		Where("hash = ? OR surrogate_id = ?", ethTx.Hash(), surrogateID.ValueOrZero()).
```

Comment:
Being extra cautious here and took a look at `models.EthTx(...)`:
```golang
func (tx Tx) EthTx(gasPriceWei *big.Int) *types.Transaction {
	return types.NewTransaction(
		tx.Nonce,
		tx.To,
		tx.Value.ToInt(),
		tx.GasLimit,
		gasPriceWei,
		tx.Data,
	)
}
```

It is a bit of a reach but I don't think `ethTx.Hash()` is unique when we have multiple accounts. ie:
1. Two accounts starting with nonce 0, configured in the default round robin
1. Both send the exact same values to the same destination with the same gas
1. This results in the same `Hash()`

Is this a problem for the `Where` clause above? It could belong to a different `jobRunId`. @se3000

Reply:
They'd have a different from address.

Reply:
If you look above, I don't believe the `From` address is used in the creation of `*types.Transaction`. I also couldn't find it being pulled into `*types.Transaction#Hash()` in geth.